### ISSUE TITLE
Missing: /opt/pkgx/ruby-lang.org/v3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.4.10

### ISSUE THREAD

**Issue by magnusviri** [pkgxdev/pantry#4010]
I'm trying to package up [gollum](https://github.com/gollum/gollum), which is written in Ruby. I don't know much about building Ruby projects. I'm hoping someone can help. Here's my package.yml and the error.

```
distributable:
  url: https://github.com/gollum/gollum/archive/refs/tags/v{{version}}.tar.gz
  strip-components: 1

versions:
  github: gollum/gollum/tags

dependencies:
  ruby-lang.org: ^3.1
  rubygems.org: '*'

build:
  dependencies:
    ruby-lang.org: ^3.1
  script: |
    bundle install
    bundle exec bin/gollum

provides:
  - bin/gollum
```

This is the error:

```
+ bundle install
/opt/pkgx/rubygems.org/v3.4.21/shim/bundle:25:in `load': cannot load such file -- /opt/pkgx/ruby-lang.org/v3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.4.10/libexec/bundle (LoadError)
	from /opt/pkgx/rubygems.org/v3.4.21/shim/bundle:25:in `<main>'
```

/opt/pkgx/ruby-lang.org/v3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.4.10 does not exist. I don't know enough to know why or what to even try to fix it.

**Comment by jhheider** [comment]
Looking at rubygems, bundler is in $PKGX_DIR/rubygems.org/gems/bundler-$version. So, possibly the RUBYLIB (or GEMPATH, or similar) path isn't correct. Basically, it's looking to the ruby directory instead of the gems directory to find bundler.

**Comment by philipgiuliani** [comment]
I'm having the same issue, after installing bundler with `gem install bundler` I get to the issue mentioned in https://github.com/pkgxdev/pantry/issues/4030#issuecomment-1807056617

**Comment by philipgiuliani** [comment]
```
❯ pkgx +ruby -- gem environment
RubyGems Environment:
  - RUBYGEMS VERSION: 3.4.22
  - RUBY VERSION: 3.2.2 (2023-03-30 patchlevel 53) [arm64-darwin22]
  - INSTALLATION DIRECTORY: /usr/local/lib/ruby/gems/3.2.0
  - USER INSTALLATION DIRECTORY: /Users/phil/.gem/ruby/3.2.0
  - RUBYGEMS PREFIX: /Users/phil/.pkgx/rubygems.org/v3.4.22
  - RUBY EXECUTABLE: /Users/phil/.pkgx/ruby-lang.org/v3.2.2/bin/ruby
  - GIT EXECUTABLE: /usr/bin/git
  - EXECUTABLE DIRECTORY: /usr/local/bin
  - SPEC CACHE DIRECTORY: /Users/phil/.gem/specs
  - SYSTEM CONFIGURATION DIRECTORY: /Users/phil/.pkgx/ruby-lang.org/etc
  - RUBYGEMS PLATFORMS:
     - ruby
     - arm64-darwin-22
  - GEM PATHS:
     - /usr/local/lib/ruby/gems/3.2.0
     - /Users/phil/.gem/ruby/3.2.0
     - /Users/phil/.pkgx/ruby-lang.org/v3.2.2/lib/ruby/gems/3.2.0
 ```

**Comment by philipgiuliani** [comment]
Can be reproduced very easily by running `pkgx bundle` wherever you want. Spent some time today trying to fix it but I can't fix it.

**Comment by jhheider** [comment]
hmmm...
```shell
$ pkgx gem install bundler
Successfully installed bundler-2.4.22
Parsing documentation for bundler-2.4.22
Done installing documentation for bundler after 0 seconds
1 gem installed
$ pkgx gem uninstall bundler
ERROR:  While executing gem ... (Errno::ENOENT)
    No such file or directory @ rb_check_realpath_internal - /usr/local/lib/ruby/gems/3.2.0
	/Users/jacob/.pkgx/rubygems.org/v3.4.22/lib/rubygems/uninstaller.rb:52:in `realpath'
	/Users/jacob/.pkgx/rubygems.org/v3.4.22/lib/rubygems/uninstaller.rb:52:in `initialize'
	/Users/jacob/.pkgx/rubygems.org/v3.4.22/lib/rubygems/commands/uninstall_command.rb:198:in `new'
	/Users/jacob/.pkgx/rubygems.org/v3.4.22/lib/rubygems/commands/uninstall_command.rb:198:in `uninstall'
	/Users/jacob/.pkgx/rubygems.org/v3.4.22/lib/rubygems/commands/uninstall_command.rb:184:in `uninstall_gem'
	/Users/jacob/.pkgx/rubygems.org/v3.4.22/lib/rubygems/commands/uninstall_command.rb:179:in `block in uninstall_specific'
	/Users/jacob/.pkgx/rubygems.org/v3.4.22/lib/rubygems/commands/uninstall_command.rb:171:in `each'
	/Users/jacob/.pkgx/rubygems.org/v3.4.22/lib/rubygems/commands/uninstall_command.rb:171:in `uninstall_specific'
	/Users/jacob/.pkgx/rubygems.org/v3.4.22/lib/rubygems/commands/uninstall_command.rb:137:in `execute'
	/Users/jacob/.pkgx/rubygems.org/v3.4.22/lib/rubygems/command.rb:328:in `invoke_with_build_args'
	/Users/jacob/.pkgx/rubygems.org/v3.4.22/lib/rubygems/command_manager.rb:253:in `invoke_command'
	/Users/jacob/.pkgx/rubygems.org/v3.4.22/lib/rubygems/command_manager.rb:193:in `process_args'
	/Users/jacob/.pkgx/rubygems.org/v3.4.22/lib/rubygems/command_manager.rb:151:in `run'
	/Users/jacob/.pkgx/rubygems.org/v3.4.22/lib/rubygems/gem_runner.rb:56:in `run'
	/Users/jacob/.pkgx/rubygems.org/v3.4.22/shim/gem:12:in `<main>'
$ pkgx gem info bundler

*** LOCAL GEMS ***

bundler (2.4.22, 2.4.10)
    Authors: André Arko, Samuel Giddins, Colby Swandale, Hiroshi
    Shibata, David Rodríguez, Grey Baker, Stephanie Morillo, Chris
    Morris, James Wen, Tim Moore, André Medeiros, Jessica Lynn Suttles,
    Terence Lee, Carl Lerche, Yehuda Katz
    Homepage: https://bundler.io
    License: MIT
    Installed at (2.4.22): /Users/jacob/.gem/ruby/3.2.0
                 (2.4.10, default): /Users/jacob/.pkgx/ruby-lang.org/v3.2.2/lib/ruby/gems/3.2.0

    The best way to manage your application's dependencies
$ rm -rf ~/.gem
$ pkgx bundle
/Users/jacob/.pkgx/rubygems.org/v3.4.22/shim/bundle:25:in `load': cannot load such file -- /Users/jacob/.pkgx/ruby-lang.org/v3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.4.10/libexec/bundle (LoadError)
	from /Users/jacob/.pkgx/rubygems.org/v3.4.22/shim/bundle:25:in `<main>'
```
so, something in our shim needs work:
```ruby
#!/usr/bin/env ruby
#
# This file was generated by RubyGems.
#
# The application 'bundler' is installed as part of a gem, and
# this file is here to facilitate running it.
#

require 'rubygems'

version = ">= 0.a"

str = ARGV.first
if str
  str = str.b[/\A_(.*)_\z/, 1]
  if str and Gem::Version.correct?(str)
    version = str
    ENV['BUNDLER_VERSION'] = str

    ARGV.shift
  end
end

if Gem.respond_to?(:activate_bin_path)
load Gem.activate_bin_path('bundler', 'bundle', version)
else
gem "bundler", version
load Gem.bin_path("bundler", "bundle", version)
end
```

**Comment by mxcl** [comment]
The original issue here is just that rubygems was not an included depedency most likely, the other issues I will investigate today

**Comment by magnusviri** [comment]
I thought this code made rubygems a build and runtime dependency.

```
dependencies:
  ruby-lang.org: ^3.1
  rubygems.org: '*'
```

Just to make sure I added it as a build dependency and I still get the error.

**Comment by mxcl** [comment]
Oh my bad, yes you are right.

**Comment by mxcl** [comment]
Fixed in #4593.

I tried to make Gollum work but I don't think you can with bundle as this creates a gem structure that must be run against the src checkout.

We are better off using the same `gem install` mechanism we use for other Ruby projects.
